fvm/environment/facade_env.go

```diff
@@ -288,6 +290,7 @@ func (env *facadeEnvironment) Reset() {
 	env.ContractUpdater.Reset()
 	env.EventEmitter.Reset()
 	env.AccountFreezer.Reset()
+	env.InterpreterSharedStateHandler.Reset()
```

Comment:
When is this function called? The interpreter shared state can currently only be re-used after a transaction ran, e.g.for contract invocations; not across transactions.

Reply:
the facadeEnv as a whole only lives as long as a transaction/script. The `Reset` method is called if transaction execution fails. In which case we have to reset everything, and then deduct fees.